File: src/main/java/io/github/gabbloquet/tddtraining/FizzBuzz/FizzBuzz.java
package io.github.gabbloquet.tddtraining.FizzBuzz;

public class FizzBuzz {
  public String convert(int input) {
    StringBuilder result = new StringBuilder();

    if (input % 3 == 0) result.append("Fizz");
    if (input % 5 == 0) result.append("Buzz");

    return result.isEmpty() ? String.valueOf(input) : result.toString();
  }

  public String compute(int to) {
    if (to < 1) throw new NonCompliantNumberException();

    StringBuilder transformedString = new StringBuilder();
    for (int i = 1; i <= to; i++) {
      transformedString.append(convert(i));
    }

    return transformedString.toString();
  }
}

File: FiZZBuZZ_Jafjaf_Sajia_PHP/src/FizzBuzz/FizzBuzz.php
<?php

namespace FizzBuzz;

class FizzBuzz
{
    public function convert(int $input): string
    {
        if ($input < 1) {
            throw new NonCompliantNumberException();
        }

        if ($input % 15 === 0) return 'FizzBuzz';
        if ($input % 5 === 0) return 'Buzz';
        if ($input % 3 === 0) return 'Fizz';

        return strval($input);
    }

    public function compute(int $to): string
    {
        if ($to < 1) {
            throw new NonCompliantNumberException();
        }

        $result = '';
        for ($i = 1; $i <= $to; $i++) {
            $result .= $this->convert($i);
        }

        return $result;
    }
}
?>